Detections/Box/BoxSensitiveFile.yaml

```diff
@@ -0,0 +1,30 @@
+id: 266746ae-5eaf-4068-a980-5d630f435c46
+name: Box - File containing sensitive data
+description: |
+  'Detects files which potentialy may contain sensitive data such as passwords, authentication tokens, secret keys.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: BoxDataConnector
+    dataTypes:
+      - BoxEvents_CL
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+  - Persistence
+query: |
+  BoxEvents
+  | where SourceItemName == 'id_rsa' or SourceItemName contains 'password' or SourceItemName contains 'key' or SourceItemName contains '_key' or SourceItemName contains '.ssh' or SourceItemName endswith '.npmrc' or SourceItemName endswith '.muttrc' or SourceItemName contains 'config.json' or SourceItemName contains '.gitconfig' or SourceItemName endswith '.netrc' or SourceItemName endswith 'package.json' or SourceItemName endswith 'Gemfile' or SourceItemName endswith 'bower.json' or SourceItemName endswith 'config.gypi' or SourceItemName endswith 'travis.yml' or SourceFileName == 'id_rsa' or SourceFileName contains 'password' or SourceFileName contains 'key' or SourceFileName contains '_key' or SourceFileName contains '.ssh' or SourceFileName endswith '.npmrc' or SourceFileName endswith '.muttrc' or SourceFileName contains 'config.json' or SourceFileName contains '.gitconfig' or SourceFileName endswith '.netrc' or SourceFileName endswith 'package.json' or SourceFileName endswith 'Gemfile' or SourceFileName contains 'bower.json' or SourceFileName contains 'config.gypi' or SourceFileName contains 'travis.yml'

```

Comment:
contains is not recommended, try using `has_any` - https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/datatypes-string-operators

Reply:
updated

Reply:
does not look like this was updated

Reply:
has_any is not applicable in this case. If we use has_any the rule will not cover all desired events(parts of file name)